src/aoc2020/day1.clj

```diff
@@ -1,62 +1,113 @@
 (ns aoc2020.day1
-  (:require [clojure.java.io :as io]
-            [clojure.string :as str]))
-
-(defn file->seq
-  [filename]
-  (-> filename
-      (io/resource)
-      (slurp)
-      (str/split-lines)))
-;;Q1: find 2 numbers in the file that would sum up to be 2020, and find the product of these 2 numbers.
-
-;Approach 1: brute force
-;loop through the entries and use 2020 to minus each number, and loop through the rest of the vectors to find the
-;if the (2020 - num) exist. Time: O(n^2) space: O(1)
-
-
-;Approach 2: binary search
-;sort the vector in ascending order, and use 2 pointers pointing to head and end of the vector, and move pointers
-;based on the sum of 2 numbers. Time: O(log n) space: O(1)
-
-(defn binary-search
-  [data-set]
-  (->> map (fn [left-pointer
-                right-pointer]
-             ((if (== 2020 (+ (get data-set left-pointer)
-                              (get data-set right-pointer)))
-
-                )))
-       (range 0 (-> data-set count))
-       (range (-> data-set count) 0 -1))
-  )
+  (:require [aoc2020.util :refer [file->seq]]))
+
+(defn find-complement
+  [sum x]
+  (- sum x))
+
+(defn compl-exist?
+  [num sum data-set]
+  (contains? data-set (find-complement sum num)))
+
+(defn filter-compl-in-dataset
+  [sum data-set]
+  (let [compl-exist? (fn [x] (compl-exist? x sum data-set))
+        result       (filterv compl-exist? data-set)
+        found?       (first result)]
+
+    (when found?
+      result)))
+
+(defn two-sum-product
+  [sum data-s]
+  (let [data-set (into #{} data-s)
+        result   (filter-compl-in-dataset sum data-set)
+        num1     (get result 0)
+        num2     (get result 1)]
+    (* num1 num2)))
+
+
+(defn compl-product
+  [num sum data-s]
+  (let [data-set   (into #{} data-s)
+        compl      (find-complement sum num)
+        result     (filter-compl-in-dataset compl data-set)
+        found?     (first result)
+        compl-data (take 2 result)]
+
+    (when found?
+      (reduce * num compl-data))))
+
+
+(defn three-sum-product
+  [sum data-s]
+  (->> (mapv
+         (fn [x] (compl-product x sum data-s))
+         data-s)
+    (filterv int?)
+    (first)))
 
 
 (comment
-  (def entries (file->seq "aoc2020/day1/input.txt"))
-  #_=>
-  ["1348"
-   "1621"
-   "1500"
-   "1818"
-   "1266"
-   ...]
-
-  (def sorted-entries (->> entries
-                           (map (fn [string] (Integer/parseInt string)))
-                           (sort)))
-  #_=>
-  (183
-    455
-    470
-    651
-    667
-    695
-    934
-    974
-    988
-    ...)
+  ;------------------------------data----------------------------------------
+
+  (do (def sample-entry-s (->> (file->seq "aoc2020/day1/input-sample.txt")
+                            (mapv parse-long)))
+    sample-entry-s)
```

Comment:
There's something wrong with your indentation. Check in with Leo (we review it yesterday) to make sure you are using your indent hotkey.